gremlin-go/driver/connection_test.go

```diff
@@ -149,266 +156,343 @@ func getEnvOrDefaultBool(key string, defaultValue bool) bool {
-	if len(value) != 0 {
-		intValue, err := strconv.Atoi(value)
-		if err == nil {
-			return intValue
-		}
+func getBasicAuthInfo() *AuthInfo {
+	return BasicAuthInfo(getEnvOrDefaultString("GREMLIN_GO_BASIC_AUTH_USERNAME", "stephen"),
+		getEnvOrDefaultString("GREMLIN_GO_BASIC_AUTH_PASSWORD", "password"))
+}
+
+func skipTestsIfNotEnabled(t *testing.T, testSuiteName string, testSuiteEnabled bool) {
+	if !testSuiteEnabled {
+		t.Skip(fmt.Sprintf("Skipping %s because %s tests are not enabled.", t.Name(), testSuiteName))
 	}
-	return defaultValue
 }
 
 func TestConnection(t *testing.T) {
-	testHost := getEnvOrDefaultString("GREMLIN_SERVER_HOSTNAME", "localhost")
-	testPort := getEnvOrDefaultInt("GREMLIN_SERVER_PORT", 8182)
-	runIntegration := getEnvOrDefaultBool("RUN_INTEGRATION_TESTS", true)
+	// Integration test variables.
+	testNoAuthUrl := getEnvOrDefaultString("GREMLIN_SERVER_URL", "ws://localhost:8182/gremlin")
+	testNoAuthEnable := getEnvOrDefaultBool("RUN_INTEGRATION_TESTS", true)
+	testNoAuthAuthInfo := &AuthInfo{}
+	testNoAuthTlsConfig := &tls.Config{}
+
+	// Basic authentication integration test variables.
+	testBasicAuthUrl := getEnvOrDefaultString("GREMLIN_SERVER_BASIC_AUTH_URL", "wss://localhost:8183/gremlin")
+	testBasicAuthEnable := getEnvOrDefaultBool("RUN_BASIC_AUTH_INTEGRATION_TESTS", false)
+	testBasicAuthAuthInfo := getBasicAuthInfo()
+	testBasicAuthTlsConfig := &tls.Config{InsecureSkipVerify: true}
+
+	t.Run("Test createConnection without valid server", func(t *testing.T) {
+		connection, err := createConnection(invalidHostValidPortValidPath, testNoAuthAuthInfo, testNoAuthTlsConfig, newLogHandler(&defaultLogger{}, Info, language.English))
+		assert.NotNil(t, err)
+		assert.Nil(t, connection)
+	})
 
-	t.Run("Test DriverRemoteConnection GraphTraversal", func(t *testing.T) {
-		if runIntegration {
-			remote, err := NewDriverRemoteConnection(testHost, testPort)
-			assert.Nil(t, err)
-			assert.NotNil(t, remote)
-			g := Traversal_().WithRemote(remote)
-
-			// Drop the graph and check that it is empty.
-			dropGraph(t, g)
-			readCount(t, g, "", 0)
-			readCount(t, g, testLabel, 0)
-			readCount(t, g, personLabel, 0)
-
-			// Add data and check that the size of the graph is correct.
-			addTestData(t, g)
-
-			readCount(t, g, "", len(testNames))
-			readCount(t, g, testLabel, 0)
-			readCount(t, g, personLabel, len(testNames))
-
-			// Read test data out of the graph and check that it is correct.
-			readTestDataVertexProperties(t, g)
-			readTestDataValues(t, g)
-
-			// Drop the graph and check that it is empty.
-			dropGraph(t, g)
-			readCount(t, g, "", 0)
-			readCount(t, g, testLabel, 0)
-			readCount(t, g, personLabel, 0)
-		}
+	t.Run("Test createConnection without valid port", func(t *testing.T) {
+		connection, err := createConnection(validHostInvalidPortValidPath, testNoAuthAuthInfo, testNoAuthTlsConfig, newLogHandler(&defaultLogger{}, Info, language.English))
+		assert.NotNil(t, err)
```

Comment:
it would be nice to assert that we got the correct error

Reply:
Not at this time - we will add this later.